src/System.CommandLine.Subsystems.Tests/HelpSubsystemTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Xunit;
+using System.CommandLine.Parsing;
+
+namespace System.CommandLine.Subsystems.Tests
+{
+    public class HelpSubsystemTests
+    {
+        [Fact]
+        public void When_help_subsystem_is_used_the_help_option_is_added_to_each_command_in_the_tree()
+        {
+            var rootCommand = new CliRootCommand
+                {
+                    new CliOption<bool>("-x"), // add option that is expected for the test data used here
+                    new CliCommand("a")
+                };
+            var configuration = new CliConfiguration(rootCommand);
+
+            var pipeline = Pipeline.CreateEmpty();
+            pipeline.Help = new HelpSubsystem();
+
+            // Parse is used because directly calling Initialize would be unusual
+            var result = pipeline.Parse(configuration, "");
+
+            rootCommand.Options.Should().NotBeNull();
+            rootCommand.Options
```

Comment:
We have a `SymbolByName` method in flight and need to ensure that it will work in this context.

Reply:
Didn't knew about this one, will add it in the upcoming commit.